carbon-route.html

```diff
@@ -271,48 +276,73 @@
         tailPath = '/' + tailPath;
       }
 
-      this._skipMatch = true;
       this._matched = matched;
-      this.data = namedMatches;
+
       var tailPrefix = this.route.prefix + matched;
 
-      if (!this.tail || this.tail.prefix !== tailPrefix || this.tail.path !== tailPath) {
-        this.tail = {
+
+      // HACK(rictic): skirting around 1.0's lack of a setMulti by poking at
+      //     internal data structures. I would not advise that you copy this
+      //     example.
+      //
+      //     This is juggling footguns.
+      this._propertySetter('data', namedMatches);
+
+      var tailChanged =
+          !this.tail ||
+          this.tail.prefix !== tailPrefix ||
+          this.tail.path !== tailPath;
+      if (tailChanged) {
+        this._propertySetter('tail', {
           prefix: tailPrefix,
           path: tailPath,
           __queryParams: this.route.__queryParams
-        };
+        });
+      }
+
+      this._dataInUrl = {};
+      for (var key in namedMatches) {
+        this._dataInUrl[key] = namedMatches[key];
       }
+
       this._setActive(true);
-      this._skipMatch = false;
+      if (tailChanged) {
+        this._pathEffector('tail', this.tail);
+        this._notifyPathUp('tail', this.tail);
+      }
+      this._pathEffector('data', this.data);
+      this._notifyPathUp('data', this.data);
```

Comment:
I wonder if this might read a little cleaner if we implemented a `setMulti` helper, e.g.,

```javascript
__setMulti(setObj) {
  for (var property in setObj) {
    this._propertySetter(property, setObj[property]);
  }

  for (var property in setObj) {
    this._pathEffector(property, this[property]);
    this._notifyPathUp(property, this[property]);
  }
}
```

Some of this conditional checking might look a bit cleaner (just check for whether or not you need to add properties to `setObj` before you call `__setMulti`).

I'm not sure if this works well in concert with all of the things in between though. I looks like updating `_dataInUrl` might be moveable. I'm not sure about updating `active` though.

WDYT?